Ignore migrations during linting

<!-- markdownlint-disable-next-line first-line-heading -->
## FEATURE DESCRIPTION

### Feature Inspiration

The changes required for black, flake8, and pylint are generally minor, but can be annoying since the code in migrations is usually generated by Django itself.

> @jcmatese Looks like there is a missing migration (forgot to check it in perhaps?)
> 
> > I give up on the linting issues, because Django migrations and Flake8 will always run afoul of each other, when it comes to line length.
> 
> This is exactly why Rob (and I) had initially suggest we _not_ lint the generated code. If you now agree that linting the migrations is more trouble that it's worth, I'd happily agree. Feel free to submit a PR or comment your approval here and I'll do it.

Yes, that is fine by me.  I am tired of fighting with it.  Skip migrations green light from me.

_Originally posted by @jcmatese in https://github.com/Princeton-LSI-ResearchComputing/tracebase/pull/55#issuecomment-828793902_

### Feature Description

Avoid running linters on generated code in the `migrations` folder.

Add exclusions to `flake8` for `.venv` and `migrations` directories.

### Alternatives Considered

None

### Comment

Add any other context or screenshots about the feature request here.

-----

## ISSUE OWNER SECTION

### Assumptions

- code in the migrations directory is generated by Django

### Requirements

- The GitHub Action for Super-Linter does not run the linters on generated code in the migrations directory

### Limitations

- Users who run linters run separately from Super-Linter will need to ignore any errors flagged for files in the migrations directory

### Affected/Changed Components

- `.github/workflows/superlinter.yml`

### DESIGN

#### GUI Change description

None

#### Code Change Description *(Pseudocode optional)*

Update the `FILTER_REGEX_EXCLUDE` variable to skip the `migrations` subdirectory.

#### Tests

Create a linting issue in one of the migrations files and see that Super-Linter ignores it after the change
No automated tests will be added

My pet peeve is that I have to reformat long lines twice, once in the model, and once again in the makemigrations output of that model, typically because of help_text definitions.  For example, again today 

```
(.venv) jcmatese@vpn12-client-172-20-205-67:~/dev/tracebase(add-PeakGroupSet) $ flake8 .
./DataRepo/migrations/0001_reinitialize.py:127:121: E501 line too long (155 > 120 characters)
./DataRepo/migrations/0001_reinitialize.py:253:121: E501 line too long (143 > 120 characters)

```

It is not high priority or even difficult, just super annoying that I find myself reformatting the same help_text to meet <120 flake8 complaints of Django formatted output.

--- a/.flake8
+++ b/.flake8
@@ -1,2 +1,3 @@
 [flake8]
 max-line-length = 120
+extend-exclude = .venv, migrations

--- a/.github/workflows/superlinter.yml
+++ b/.github/workflows/superlinter.yml
@@ -25,4 +25,4 @@ jobs:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           LINTER_RULES_PATH: /
           IGNORE_GITIGNORED_FILES: true
-          FILTER_REGEX_EXCLUDE: (\.pylintrc)
+          FILTER_REGEX_EXCLUDE: (\.pylintrc|migrations)

